@here/harp-materials/lib/MapMeshMaterials.ts

```diff
@@ -1230,6 +1240,129 @@ export class MapMeshStandardMaterial extends THREE.MeshStandardMaterial
     // Mixin declarations end -----------------------------------------------------------
 }
 
+/**
+ * Subclass of THREE.StandardMaterial. Uses the standard material (used in
+ * MapMeshStandardMaterial) as the base and removes the diffuse / specular components of the
+ * lighting to keep the colors consistent (that aren't in shadow).
+ *
+ * @see [[Tile#addRenderHelper]]
+ */
+export class MapMeshFlatStandardMaterial extends THREE.ShaderMaterial {
```

Comment:
Should't it derive from MapMeshBasicMaterial and just add the shadow chunk?

Reply:
I thought that as well, but there is much more similarity with the standard material than the basic one, because the standard material already has the correct structure, it just requires changing this one part. Otherwise I would need to pick out the correct #includes (they also depend on each other also), I would need to set the correct uniforms for the lights.

I can look into this if you think it is important, but it will take more time.

Reply:
But why does it derive from ShaderMAterial. It should derive from MapStandardMaterial then (i.e. THREE.MeshStandardMaterial + Fading).

Reply:
I tried this originally, it doesn't work because threejs has a shader cache internally, and if I modify the shader code, all MapMeshStandardMaterials are modified, see: https://github.com/mrdoob/three.js/blob/bcf0aca489b2a8e1da2b795fffcb36bb379ba665/src/renderers/webgl/WebGLPrograms.js#L55

Reply:
> I tried this originally, it doesn't work because threejs has a shader cache internally

@nzjony It _is_ possible to fix only this particular chunk. However it's tricky as hell. `getShaderObject` is followed by `onBeforeCompile` when you should "patch" shader like we do. We have many examples of this (grep onBeforeCompile in MapMeshMaterials).

Or ... you've found some fundamental problem and all of our "patching" machinery doesn't work.
@kkostecki did lots of work cleaning this up, he may help.

Reply:
Good point, I will investigate, I got some weird artifacts, and thought it wasn't possible,. because I noticed the shader was being changed for other shaders of the same type, but yes of course, it should be possible. I will retry.

Reply:
I looked into it, I wasn't differentiating between extruded-polygons and the fill technique (which both internally use the standard material from three.js, I read the three.js code wrong, so taking me down the wrong path, thanks @ninok  and @zbigg  for the comment in this regard!)